### ISSUE TITLE
[API] Application History: extend endpoint to record state change info

### ISSUE THREAD

**Issue by kcullion** [icgc-argo/dac-api#226]
## Goal

Extend the GET application by ID endpoint:  https://dac.qa.argo.cancercollaboratory.org/api/api-docs/#/Application/get_applications__id_to show the new values in the `updates` section with record the mini-history of the fields in mockup
![image](https://user-images.githubusercontent.com/18424138/135472658-431d0654-68a0-4978-88c0-e6598b45b4b9.png)

Related UI tickets
https://github.com/icgc-argo/dac-ui/issues/470
https://github.com/icgc-argo/dac-ui/issues/471

## Expected Outcomes
Need to record actions for each STATE CHANGE:
- [ ] Record **WHEN** the action was done (Date of Status Change)
- [ ] Record **WHO** did the action based on role + id  (Action Performed By)
- [ ] Calculate the days from last state change (to be used in the tsv report ticket:https://github.com/icgc-argo/dac-api/issues/227 )
New calculated field **Application Type** based on isRenewal value
For application renewal we need a new migration on the CREATE endpoint
- [ ] Set isRenewal = false (or "New")
- [ ] Set ableToRenew = false

- [ ] Migration for existing applications to support existing states.  Note: about the migration for the updates section: we will only be able to add the current state as the first update event

## From meeting with Bashar Oct 14
- To simplify we will just sort by date of status/event change. 
- Let's do it the cheap way for now - not a new collection, but just a new response type that grabs all updates from the applications and then does the sort

## For testing
- This will only work for new applications going forward. Ignore the old apps

**Comment by kcullion** [comment]
@anncatton  this works great! Well done! With a few new applications I tested:

DACO-252, I see the correct status changes in the modal: 
1. When I created an application
2. When I submitted the application
3. When admin request revisions
4. When I fixed the application and resubmitted for review
5. When admin approved the application
6. When I close the application the next day, the modal data was correct

DACO-255 -
I submitted and Requested revisions twice and then rejected - app history was correct. Did this across multiple days and the date was correct.

DACO-256
I tried closing before submitting - this reflects in modal correctly.
